{
  "active_lock_reason": null,
  "assignee": null,
  "assignees": [],
  "author_association": "CONTRIBUTOR",
  "body": "<!-- This form is for bug reports and feature requests ONLY!\r\n\r\nIf you're looking for help check [Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes) and the [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/).\r\n\r\nIf the matter is security related, please disclose it privately via https://kubernetes.io/security/.\r\n-->\r\n\r\n**Is this a BUG REPORT or FEATURE REQUEST?**:\r\n\r\n/kind bug\r\n/sig testing\r\n\r\n**What happened**:\r\n\r\nThe test ``[sig-api-machinery] AdmissionWebhook Should be able to deny attaching pod`` can randomly fail because the test never waits for the pod to get into a ``Running`` state before trying to execute ``kubectl attach``. This can happen more frequently when it has to pull the necessary image. Additionally, spawning and starting pods takes a bit longer on Windows nodes.\r\n\r\n    \u00e2\u00a2 Failure [88.631 seconds]\r\n    [sig-api-machinery] AdmissionWebhook\r\n    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/framework.go:22\r\n      Should be able to deny attaching pod [It]\r\n      /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/webhook.go:125\r\n\r\n      Oct  8 16:36:05.987: unexpected 'kubectl attach' error message. expected to contain \"attaching to pod 'to-be-attached-pod' is not allowed\", got \"error running &{/usr/local/bin/kubectl [kubectl --server=https://k8s-cbelu-6.eastus.cloudapp.azure.com --kubeconfig=/home/ubuntu/kube/config_6 attach --namespace=e2e-tests-webhook-d8clc to-be-attached-pod -i -c=container1] []  <nil>  If you don't see a command prompt, try pressing enter.\\nerror: unable to upgrade connection: container container1 not found in pod to-be-attached-pod_e2e-tests-webhook-d8clc\\n [] <nil> 0xc0018941b0 exit status 1 <nil> <nil> true [0xc0002d1808 0xc0002d1838 0xc0002d1860] [0xc0002d1808 0xc0002d1838 0xc0002d1860] [0xc0002d1820 0xc0002d1850] [0x92bfb0 0x92bfb0] 0xc00074dd40 <nil>}:\\nCommand stdout:\\n\\nstderr:\\nIf you don't see a command prompt, try pressing enter.\\nerror: unable to upgrade connection: container container1 not found in pod to-be-attached-pod_e2e-tests-webhook-d8clc\\n\\nerror:\\nexit status 1\\n\"\r\n\r\nAdditionally, the ``kubectl attach`` command could hang (if it actually managed to attach), causing the test to never end. The command should be executed with a timeout.\r\n\r\n**What you expected to happen**:\r\n\r\nThe test should wait for the pod to enter a ``Running`` state before trying to attach to it, so it won't randomly fail.\r\nThe test should have a timeout when trying to attach to the pod.\r\n\r\n**How to reproduce it (as minimally and precisely as possible)**:\r\n\r\nRun the ``[sig-api-machinery] AdmissionWebhook Should be able to deny attaching pod`` test using Windows nodes and / or delete webhook image from the Compute Node's image cache.\r\n\r\n**Anything else we need to know?**:\r\n\r\n**Environment**:\r\n- Kubernetes version (use `kubectl version`): v1.12.0-rc.2\r\n- Cloud provider or hardware configuration: N/A\r\n- OS (e.g. from /etc/os-release): Windows Server 1803 compute nodes\r\n- Kernel (e.g. `uname -a`): N/A\r\n- Install tools: N/A\r\n- Others:\r\n",
  "closed_at": "2018-10-12T02:35:42Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 0,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/69567/comments",
  "created_at": "2018-10-09T11:28:08Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/69567/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/69567",
  "id": 368163879,
  "labels": [
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Testing.",
      "id": 483069764,
      "name": "sig/testing",
      "node_id": "MDU6TGFiZWw0ODMwNjk3NjQ=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/testing"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/69567/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWUzNjgxNjM4Nzk=",
  "number": 69567,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "webhook test can randomly fail because it never waits for the pod to be Running",
  "updated_at": "2018-10-12T02:35:42Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/69567",
  "user": {
    "avatar_url": "https://avatars2.githubusercontent.com/u/1552519?v=4",
    "events_url": "https://api.github.com/users/claudiubelu/events{/privacy}",
    "followers_url": "https://api.github.com/users/claudiubelu/followers",
    "following_url": "https://api.github.com/users/claudiubelu/following{/other_user}",
    "gists_url": "https://api.github.com/users/claudiubelu/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/claudiubelu",
    "id": 1552519,
    "login": "claudiubelu",
    "node_id": "MDQ6VXNlcjE1NTI1MTk=",
    "organizations_url": "https://api.github.com/users/claudiubelu/orgs",
    "received_events_url": "https://api.github.com/users/claudiubelu/received_events",
    "repos_url": "https://api.github.com/users/claudiubelu/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/claudiubelu/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/claudiubelu/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/claudiubelu"
  }
}